Fix the issue that controller is not started

jiaqiluo opened the pull request:
Issue: https://github.com/rancher/rancher/issues/36518

Problem:
If we create an RKE/RKE2 downstream cluster with multiple control plane nodes and with ClusterAuthEndpoint enabled, then use the secondary contexts in the kubeconfig file downloaded from Rancher UI to access the cluster, we will hit the following error with one of the secondary contexts:
```
error: You must be logged in to the server (Unauthorized)
```
meanwhile, the logs of the `kube-api-auth` pod on this control plane show the following kind of error:
```
time="2022-02-24T21:11:46Z" level=info msg="  ...looking up token for kubeconfig-user-qmk2kdmjlt"
time="2022-02-24T21:11:46Z" level=error msg="clusteruserattributes.cluster.cattle.io \"cattle-system/user-qmk2k\" not found"
```
It indicates the controller for ClusterUserAttribute is not started.

Solutions:
Sometimes the ClusterUserAttribute controller is not started, which causes the failure to retrieve the ClusterUserAttribute for the current user and subsequently the failure to validate the cluster auth token. Lasso has the behavior that it will delay the start of a controller if there is no listener or handler associated with it, and in our case, it might cause the controller never to start. 
So the fix is to move the registration of listeners before starting the controller, which makes sure the controller is started immediately.

Tests:
Both RKE and RKE2 clusters are tested on a Rancher setup with fixes using the customized `kube-api-auth` image. 


TODO After merging:

- [x] cut a new tag
- [ ] update rancher/rancher to use the new tag

jiaqiluo committed:
bump dependencies

thedadams reviewed (commented):
Does this fix the underlying issue?

jiaqiluo committed:
Sometimes the ClusterUserAttribute controller is not started, which causes the failure to retrieve the ClusterUserAttribute for the current user and subsequently the failure to validate the cluster auth token.
Lasso has the behavior that it will delay the start of a controller if there is no listener or handler associated with it, and in our case, it might cause the controller never to start.
So the fix is to move the registration of listeners before starting the controller, which makes sure the controller is started immediately.

thedadams commented:
@jiaqiluo Can you update the description, please?
